src/Stoa.ts

```diff
@@ -21,11 +23,31 @@ class Stoa {
             res.send(req.params.address + ':' + req.query.height);
         });
 
+        /**
+         * When a request is received through agora-history push
+         * JSON block data is parsed and stored on each storage.
+         */
         this.stoa.post("/push",
-            (req: express.Request, res: express.Response, next: express.NextFunction) => {
-            console.log(req.body);
-            res.status(200).send();
-        });
+            asyncHandler(async (req: express.Request, res: express.Response, next: express.NextFunction) => {
+            var data: any = req.body;
+            var block: any = JSON.parse(data.block);
```

Comment:
If parsing fails, what about sending 500 and logging the error?

Reply:
Good catch! There is a possibility of an error.